AmplifyPlugins/Core/AWSPluginsCoreTests/Model/Decorator/AuthRuleDecorator/ModelWithOwnerAuthAndMultiGroup.swift

```diff
@@ -0,0 +1,242 @@
+    public var title: String
+    public var owner: String?
+
+    public init(id: String = UUID().uuidString,
+                title: String,
+                owner: String? = nil) {
+        self.id = id
+        self.title = title
+        self.owner = owner
+    }
+
+    public enum CodingKeys: String, ModelKey {
+        case id
+        case title
+        case owner
+    }
+
+    public static let keys = CodingKeys.self
+
+    public static let schema = defineSchema { model in
+        let oIDCMultiGroupPost = OIDCMultiGroupPost.keys
+
+        model.authRules = [
+            rule(allow: .owner,
+                 ownerField: "owner",
+                 identityClaim: "sub",
+                 operations: [.create, .update, .delete, .read]),
+            rule(allow: .groups,
+                 groupClaim: "https://myapp.com/claims/groups",
+                 groups: ["Admins"],
+                 operations: [.create, .update, .delete, .read]),
+            rule(allow: .groups,
+                 groupClaim: "https://differentapp.com/claims/groups",
+                 groups: ["Moderators", "Editors"],
+                 operations: [.create, .update, .delete, .read])
+        ]
+        model.pluralName = "OIDCMultiGroupPosts"
+        model.fields(
+            .id(),
+            .field(oIDCMultiGroupPost.title, is: .required, ofType: .string),
+            .field(oIDCMultiGroupPost.owner, is: .optional, ofType: .string)
+        )
+    }
+}
+
+class ModelWithOwnerAuthAndMultiGroup: XCTestCase {
+    override func setUp() {
+        ModelRegistry.register(modelType: OIDCMultiGroupPost.self)
+    }
+
+    override func tearDown() {
+        ModelRegistry.reset()
+    }
+
+    func testOnCreateSubscription_NoGroupInfoPassed() {
+        let claims = ["username": "user1",
+                      "sub": "123e4567-dead-beef-a456-426614174000"] as IdentityClaimsDictionary
+        var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelType: OIDCMultiGroupPost.self,
+                                                               operationType: .subscription)
+        documentBuilder.add(decorator: DirectiveNameDecorator(type: .onCreate))
+        documentBuilder.add(decorator: AuthRuleDecorator(.subscription(.onCreate, claims)))
+        let document = documentBuilder.build()
+        let expectedQueryDocument = """
+        subscription OnCreateOIDCMultiGroupPost($owner: String!) {
+          onCreateOIDCMultiGroupPost(owner: $owner) {
+            id
+            owner
+            title
+            __typename
+          }
+        }
+        """
+        XCTAssertEqual(document.name, "onCreateOIDCMultiGroupPost")
+        XCTAssertEqual(document.stringValue, expectedQueryDocument)
+        guard let variables = document.variables else {
+            XCTFail("The document doesn't contain variables")
+            return
+        }
+        XCTAssertEqual(variables["owner"] as? String, "123e4567-dead-beef-a456-426614174000")
```

Comment:
Just as I was reading these tests, i went to the top and and back here to match the claims with the schema so if we could sprinkle assert messages around, it would make that special "add owner or not" logic more clear

```suggestion
        XCTAssertEqual(variables["owner"] as? String, 
            "123e4567-dead-beef-a456-426614174000"
            "owner should exist since there were no groups present in the claims to match the schema")
```

Reply:
Good idea, i'll go back and add this.